aws/resource_aws_glue_dev_endpoint.go

```diff
@@ -0,0 +1,588 @@
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"number_of_nodes": {
+				Type:          schema.TypeInt,
+				Optional:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"number_of_workers", "worker_type"},
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					return new == "0"
+				},
+				ValidateFunc: validation.IntAtLeast(2),
+			},
+
+			"number_of_workers": {
+				Type:          schema.TypeInt,
+				Optional:      true,
+				ForceNew:      true,
+				ValidateFunc:  validation.IntAtLeast(2),
+				ConflictsWith: []string{"number_of_nodes"},
+			},
+
+			"public_key": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"public_keys"},
+			},
+
+			"public_keys": {
+				Type:          schema.TypeSet,
+				Optional:      true,
+				Elem:          &schema.Schema{Type: schema.TypeString},
+				Set:           schema.HashString,
+				ConflictsWith: []string{"public_key"},
+				MaxItems:      5,
+			},
+
+			"role_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateArn,
+			},
+
+			"security_configuration": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"security_group_ids": {
+				Type:         schema.TypeSet,
+				Optional:     true,
+				ForceNew:     true,
+				Elem:         &schema.Schema{Type: schema.TypeString},
+				Set:          schema.HashString,
+				RequiredWith: []string{"subnet_id"},
+			},
+
+			"subnet_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				RequiredWith: []string{"security_group_ids"},
+			},
+
+			"tags": tagsSchema(),
+
+			"private_address": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"public_address": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"yarn_endpoint_address": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"zeppelin_remote_spark_interpreter_port": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"worker_type": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ValidateFunc:  validation.StringInSlice(glue.WorkerType_Values(), false),
+				ConflictsWith: []string{"number_of_nodes"},
+				ForceNew:      true,
+			},
+
+			"availability_zone": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"vpc_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"failure_reason": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsGlueDevEndpointCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).glueconn
+
+	var name string
+	if v, ok := d.GetOk("name"); ok {
+		name = v.(string)
+	} else {
+		name = resource.UniqueId()
+	}
```

Comment:
not sure about this, id rather match the API and set this to required and lose the unique name generator